app/src/main/java/org/fossasia/openevent/app/event/attendees/AttendeesActivity.java

```diff
@@ -121,34 +111,39 @@ public void checkInAlertBuilder(final int index) {
             + "Ticket: " + thisAttendee.getTicket().getType())
             .setPositiveButton("OK", (dialog, which) -> {
                 Log.d(TAG, "onClick: inside ok");
-                changeCheckStatus(thisAttendee.getId(), index);
+                attendeesPresenter.toggleAttendeeCheckStatus(index, thisAttendee.getId());
             }).setNegativeButton("CANCEL", (dialog, which) -> {
 
             });
         builder.create();
         builder.show();
     }
 
-    public void changeCheckStatus(Long thisAttendeeId, final int position) {
-        if (Network.isNetworkConnected(this)) {
-            ApiCall.PostApiCall(this, Constants.EVENT_DETAILS + id + Constants.ATTENDEES_TOGGLE + thisAttendeeId, new VolleyCallBack() {
-                @Override
-                public void onSuccess(String result) {
-                    Log.d(TAG, "onSuccess: " + result);
-                    Gson gson = new Gson();
-                    Attendee newAttendeeDetails = gson.fromJson(result, Attendee.class);
-                    attendeeArrayList.set(position, newAttendeeDetails);
-                    attendeeListAdapter.notifyDataSetChanged();
-                }
-
-                @Override
-                public void onError(VolleyError error) {
-
-                }
-            });
-        } else {
-            Toast.makeText(this, Constants.NO_NETWORK, Toast.LENGTH_SHORT).show();
+    @Override
+    public void showProgressBar(boolean show) {
+        if(show) {
+            progressBar.setVisibility(View.VISIBLE);
         }
+        else{
+            progressBar.setVisibility(View.GONE);
+        }
+    }
+
+    @Override
+    public void loadAttendeesIntoList(List<Attendee> attendees) {
+        attendeeArrayList.clear();
+        attendeeArrayList.addAll(attendees);
+        attendeeListAdapter.notifyDataSetChanged();
+    }
 
+    @Override
+    public void updateAttendee(int position, Attendee attendee) {
+        attendeeArrayList.set(position, attendee);
+        attendeeListAdapter.notifyDataSetChanged();
```

Comment:
Don't use index, because it will not remain constant till the callback comes back.

For instance, if we implement sorting in future, and user sorts using different criteria before the response comes back, then this code will update the wrong attendee